app/src/main/java/org/desarrolladorslp/teckersapp/ui/messages/MessageViewModel.kt

```diff
@@ -1,18 +1,68 @@
 package org.desarrolladorslp.teckersapp.ui.messages
 
-import androidx.lifecycle.LiveData
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat.startActivity
 import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.Transformations
 import androidx.lifecycle.ViewModel
+import com.google.android.gms.auth.api.Auth
+import org.desarrolladorslp.teckersapp.MainActivity
+import org.desarrolladorslp.teckersapp.NavigationMenu
+import org.desarrolladorslp.teckersapp.exception.AuthorizationException
+import org.desarrolladorslp.teckersapp.exception.ResponseException
 
-class MessageViewModel : ViewModel() {
+import org.desarrolladorslp.teckersapp.model.Inbox
+import org.desarrolladorslp.teckersapp.model.MessageHeader
+import org.desarrolladorslp.teckersapp.service.APIEndpoint
+import org.desarrolladorslp.teckersapp.service.MessageService
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
 
-    private val _index = MutableLiveData<Int>()
-    val text: LiveData<String> = Transformations.map(_index) {
-        "Hello world from section: $it"
+class MessageViewModel() : ViewModel() {
+
+    private var messageService = APIEndpoint.instance()?.create(MessageService::class.java);
+
+    val _inbox = MutableLiveData<Inbox>()
+    val _responseException = MutableLiveData<ResponseException?>()
+    val _authorizationException = MutableLiveData<AuthorizationException?>()
+
+    fun getmessages(priorityIndex: Int): ArrayList<MessageHeader> {
+        if (priorityIndex == LOW_PRIORITY_PAGE_INDEX) {
+            return _inbox.value!!.lowPriority
+        }
+        return _inbox.value!!.highPriority
+    }
+
+    fun getInbox() {
+        var messageCall = messageService?.getMessages()
+        messageCall?.enqueue(object : Callback<Inbox> {
+            override fun onResponse(call: Call<Inbox>, response: Response<Inbox>) {
+                _inbox.value = response.body()
+            }
+
+            override fun onFailure(call: Call<Inbox>, t: Throwable) {
+                val m = t.message;
+
+                if (t is ResponseException) {
```

Comment:
Here make sense to have different behavior for those two exceptios

Reply:
okey